src/app/Services/RelationshipService.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+namespace App\Services;
+
+use App\Models\Relationship;
+class RelationshipService
+{
+    public function follow(int $userToFollowId): void
+    {
+        $existingRelationship = Relationship::where('user_id_follower', auth()->user()->id)
+            ->where('user_id_following', $userToFollowId)
+            ->onlyTrashed()
```

Comment:
I wonder why are you querying with the deleted records?

Reply:
I modified the code to use status column instead.